### PR TITLE
Hardening ftest

### PR DESCRIPTION
**Pull request by fgalan**
A couple of hardening fixes. Details in inline comments in the PR.

### CODE REVIEW

File: src/lib/cache/subCache.cpp [hunk 1]
```diff
@@ -1260,7 +1260,7 @@ extern bool noCache;
 *
 * If 'errors' == 0, then the subscription is marked as non-erroneous.
 */
-void subCacheItemNotificationErrorStatus
+void subNotificationErrorStatus
```
Comment (fgalan) [inline src/lib/cache/subCache.cpp:1263]: The `subCache` token as part of the name was misleading. The function also does its work in noCache mode (see the `if (noCache)` part inside the function body).

NTC (just informative)

File: test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_uri_param_for_metadata_nocache.test [hunk 2]
```diff
@@ -0,0 +1,867 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
```
Comment (fgalan) [inline test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_uri_param_for_metadata_nocache.test:1]: This file is a copy of the existing test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_uri_param_for_metadata.test file and adapted to noCache scenario. These are the exact differences between both:

```
$ diff test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_uri_param_for_metadata.test test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_uri_param_for_metadata_nocache.test -u
```


```diff
--- test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_uri_param_for_metadata.test	2021-03-09 08:52:56.782922095 +0100
+++ test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_uri_param_for_metadata_nocache.test	2021-03-09 08:52:56.782922095 +0100
@@ -21,16 +21,26 @@
 # VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
 
 --NAME--
-mq as URI param for metadata in filters
+mq as URI param for metadata in filters - no subCache variant
 
 --SHELL-INIT--
 dbInit CB
-brokerStart CB 0 IPv4 --cache
+brokerStart CB 0 IPv4 --noCache
 accumulatorStart --pretty-print
 
 --SHELL--
 
 #
+# The only difference regarding the cache version of this test (mq_as_uri_param_for_metadata.test)
+# is in step 05b. This test doesn't include lastSuccess and lastSucessInfo fields, while
+# mq_as_uri_param_for_metadata.test does
+#
+# In step 05 there is a race condition between the creation of the sub document
+# at DB and the notification being sent. The notification being create is the one that wins
+# (empirically checked) so the sub document doesn't exist at the DB when it tries to update
+# the lastSuccess and lastSucessInfo
+#
+#
 # 01. Create an entity E1 with attr A1 with metadata M1=12
 # 02. Create an entity E2 with attr A1 with metadata M1=13
 # 03. Create an entity E3 with attr A1 with metadata M1=14
@@ -401,7 +411,7 @@
 05b. GET subscription to see 'mq' correctly rendered
 ====================================================
 HTTP/1.1 200 OK
-Content-Length: 442
+Content-Length: 379
 Content-Type: application/json
 Fiware-Correlator: REGEX([0-9a-f\-]{36})
 Date: REGEX(.*)
@@ -416,8 +426,6 @@
             "url": "http://localhost:REGEX(\d+)/notify"
         },
         "lastNotification": "REGEX(.*)",
-        "lastSuccess": "REGEX(.*)",
-        "lastSuccessCode": 200,
         "onlyChangedAttrs": false,
         "timesSent": 1
     },
```

NTC (informative)

Comment (fgalan) [reply test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_uri_param_for_metadata_nocache.test:1]: First version has some problems. This is the definitive one:

```diff
--- test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_uri_param_for_metadata.test	2021-03-09 08:52:56.782922095 +0100
+++ test/functionalTest/cases/1156_mq_as_uri_param_for_metadata/mq_as_uri_param_for_metadata_nocache.test	2021-03-09 12:17:28.336374318 +0100
@@ -21,16 +21,27 @@
 # VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
 
 --NAME--
-mq as URI param for metadata in filters
+mq as URI param for metadata in filters - no subCache variant
 
 --SHELL-INIT--
 dbInit CB
-brokerStart CB 0 IPv4 --cache
+brokerStart CB 0 IPv4 --noCache
 accumulatorStart --pretty-print
 
 --SHELL--
 
 #
+# The only difference regarding the cache version of this test (mq_as_uri_param_for_metadata.test)
+# is in step 05b. This test doesn't include lastSuccess and lastSucessInfo fields, while
+# mq_as_uri_param_for_metadata.test does
+#
+# In step 05 there is a race condition between the creation of the csub document
+# at DB and the notification being sent. If the notification being created is the one that wins,
+# the csub document doesn't exist at the DB when it tries to update the lastSuccess and
+# lastSucessInfo. The grep -v in step 05b is for skipping these elements and make the checking
+# valir for both cases. The REGEX in Content-Lenght covers both cases.
+#
+#
 # 01. Create an entity E1 with attr A1 with metadata M1=12
 # 02. Create an entity E2 with attr A1 with metadata M1=13
 # 03. Create an entity E3 with attr A1 with metadata M1=14
@@ -147,7 +158,7 @@
 
 echo "05b. GET subscription to see 'mq' correctly rendered"
 echo "===================================================="
-orionCurl --url /v2/subscriptions/$subId
+orionCurl --url /v2/subscriptions/$subId | grep -v lastSuccess
 echo
 echo
 
@@ -401,7 +412,7 @@
 05b. GET subscription to see 'mq' correctly rendered
 ====================================================
 HTTP/1.1 200 OK
-Content-Length: 442
+Content-Length: REGEX((379|442))
 Content-Type: application/json
 Fiware-Correlator: REGEX([0-9a-f\-]{36})
 Date: REGEX(.*)
@@ -416,8 +427,6 @@
             "url": "http://localhost:REGEX(\d+)/notify"
         },
         "lastNotification": "REGEX(.*)",
-        "lastSuccess": "REGEX(.*)",
-        "lastSuccessCode": 200,
         "onlyChangedAttrs": false,
         "timesSent": 1
     },
```

**Review by AlvaroVega — APPROVED** [review]
LGTM

**Pull request merged**